src/Features/CSharp/Portable/MakeLocalFunctionStatic/PassInCapturedVariablesAsArgumentsCodeFixProvider.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(PassInCapturedVariablesAsArgumentsCodeFixProvider)), Shared]
+    internal sealed class PassInCapturedVariablesAsArgumentsCodeFixProvider : CodeFixProvider
+    {
+        //  "CS8421: A static local function can't contain a reference to <variable>."
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create("CS8421");
+
+        public override FixAllProvider? GetFixAllProvider()
+        {
+            return null;
```

Comment:
so, it really should be pretty simple to implement this.  just derive from SyntaxEditorBasedCodeFixProvider.   Then, you'll have to implement FixAllAsync.  You'll basically implement this like so:

1. extract the logic you have in RegisterCodeFixesAsync to become a boolean check if you can fix a diagnostic (probably with an out-param that is the LocalFunction to fixup).
2. you'll have a list of diagnostics.  iterate through all of them trying to get the fixable local function.
3. now process all those local functions producing hte edits against them.

Reply:
Thanks! Turns out what I missed yesterday was filtering out duplicate diagnostics (in terms of enclosing local function). Now it's working.